packages/js/plugins/filesystem/src/mutation/index.ts

```
@@ -0,0 +1,41 @@
+import {
+  Client,
+  Input_mkdir,
+  Input_rm,
+  Input_rmdir,
+  Input_writeFile,
+  Module,
+} from "./w3";
+
+import fs from "fs";
+
+export type MutationConfig = Record<string, unknown>;
+
+export class Mutation extends Module<MutationConfig> {
+  async writeFile(
+    input: Input_writeFile,
+    _client: Client
+  ): Promise<boolean | null> {
+    fs.writeFileSync(input.path, Buffer.from(input.data));
+
+    return true;
```

If we're returning a boolean instead of void, we might as well catch the error and return false in that case

---

I'd lean on throwing an error and not returning anything in that case.

---

After a call with @pileks we decided it'd be better to just let the plugin throw so that the app can catch the error instead of masking it